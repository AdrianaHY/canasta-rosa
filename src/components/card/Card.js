import React, { Component } from 'react';

import './Card.scss';

class Card extends Component {
	render() {
		const { product } = this.props;
		return (
			<div className="product-card">
				<img className="product-image" src={product.photo.small} alt={product.name} />
				<div>
					<p className="product-name">{product.name}</p>
					<p className="store-name">{product.store.name}</p>
					<p><strong>${product.price}MX</strong></p>
				</div>
			</div>
		)
	}
}

export default Card;