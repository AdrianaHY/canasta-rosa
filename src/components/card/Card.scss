@import '../../styles/_variables.scss';

.product-card {
	font-family: $fontFamily;
	border: 1px solid $lightGray;
	border-radius: 5px;
	max-width: 300px;
	padding: $paddingSmall;
	margin: $paddingLarge;

	.product-image {
		max-width: 300px;
	}

	.product-name {
		font-size: 18px;
	}

	.store-name {
		color: $gray;
	}
}