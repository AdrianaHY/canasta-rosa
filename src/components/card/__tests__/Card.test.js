import React from 'react';
import { shallow } from 'enzyme';
import Card from '../Card';

describe('components | card | Card', () => {
	it('Renders the component', () => {
		const props = {
			product: {
				name: 'product',
				photo: {
					small: 'small.png'
				},
				store: {
					name: 'store'
				},
				price: 45
			}
		}

		const component = shallow(
			<Card {...props}/>
		);

		expect(component).toMatchSnapshot();
	})
})