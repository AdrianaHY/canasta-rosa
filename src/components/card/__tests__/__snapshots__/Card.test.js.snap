// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | card | Card Renders the component 1`] = `ShallowWrapper {}`;
